linera-sdk/src/service/runtime.rs

```diff
@@ -33,6 +70,57 @@ impl ServiceRuntime {
         Self::fetch_value_through_cache(&self.next_block_height, || wit::next_block_height().into())
     }
 
+    /// Retrieves the current system time, i.e. the timestamp of the block in which this is called.
+    pub fn system_time(&self) -> Timestamp {
+        Self::fetch_value_through_cache(&self.timestamp, || wit::read_system_timestamp().into())
+    }
+
+    /// Returns the current chain balance.
+    pub fn chain_balance(&self) -> Amount {
+        Self::fetch_value_through_cache(&self.chain_balance, || wit::read_chain_balance().into())
+    }
+
+    /// Returns the balance of one of the chain owners.
```

Comment:
But they aren't necessarily a chain owner!

Reply:
Oops! Fixed in `ContractRuntime` but forgot to fix here as well. Now it's done :sweat_smile: